atom/browser/native_window_views_win.cc

```diff
@@ -207,4 +220,86 @@ void NativeWindowViews::HandleSizeEvent(WPARAM w_param, LPARAM l_param) {
   }
 }
 
+void NativeWindowViews::SetForwardMouseMessages(bool forward) {
+  if (forward && !forwarding_mouse_messages_) {
+    forwarding_mouse_messages_ = true;
+    forwarding_windows_.insert(this);
+    
+    // Subclassing is used to fix some issues when forwarding mouse messages;
+    // see comments in |SubclassProc|.
+    SetWindowSubclass(
+      legacy_window_, SubclassProc, 1, reinterpret_cast<DWORD_PTR>(this));
+
+    if (!mouse_hook_) {
+      mouse_hook_ = SetWindowsHookEx(WH_MOUSE_LL, MouseHookProc, NULL, 0);
+    }
+  } else if (!forward && forwarding_mouse_messages_) {
+    forwarding_mouse_messages_ = false;
+    forwarding_windows_.erase(this);
```

Comment:
The cleanup is only done when SetForwardMouseMessages(false) is called, it should also be done when the window is destroyed.